--- etc/init.d/rc.voip	2005-08-06 16:05:29.000000000 +0200
+++ etc/init.d/rc.voip	2005-08-06 16:35:55.000000000 +0200
@@ -6,11 +6,33 @@
   "" | start ) 
 
        if [ "`pidof telefon`" = "" ] ; then
-         if [ "$AB_TEST" = "1" ] ; then
-            telefon > /dev/null
-         else
-            telefon a127.0.0.1> /dev/null
-         fi
+	get_it() {
+	    local VALUE="$(sed -ne "s/^$1=\(['\"]\?\)\(.*\)\1[        ]*/\2/p" \
+		/var/flash/debug.cfg)"
+	    echo "${VALUE:-"$2"}"
+	}
+	CALLMONITOR="$(get_it CALLMONITOR 0)"
+	CALLMONITOR_OPTIONS="$(get_it CALLMONITOR_OPTIONS)"
+	TELEFON_LOCAL="$(get_it TELEFON_LOCAL 1)"
+	TELEFON_QUIET="$(get_it TELEFON_QUIET 1)"
+
+	if [ ! -x /usr/bin/callmonitor ]; then CALLMONITOR=0; fi
+	if [ "$TELEFON_LOCAL" = 1 ]; then
+	    TELEFON="telefon a127.0.0.1"
+	else
+	    TELEFON="telefon"
+	fi
+	if [ "$TELEFON_QUIET" = 1 ]; then
+	    REDIR="> /dev/null"
+	else
+	    REDIR=
+	fi
+	if [ "$CALLMONITOR" = 1 ]; then
+	    MONITOR="| callmonitor $CALLMONITOR_OPTIONS"
+	else
+	    MONITOR=
+	fi
+	eval "$TELEFON $MONITOR $REDIR &"
        fi
        if [ "`pidof voipd`" = "" ] ; then
           voipd
